opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -92,8 +108,21 @@ impl Resource {
         KV: IntoIterator<Item = KeyValue>,
         S: Into<Cow<'static, str>>,
     {
+        let schema_url_str = schema_url.into();
```

Comment:
let schema_url_str = schema_url.into();
 let normalized_schema_url = if schema_url_str.is_empty() {
            None
        } else {
            Some(schema_url_str)
        };
        let mut attrs = HashMap::new();
        for kv in kvs {
            attrs.insert(kv.key, kv.value);
        }

        Resource {
            inner: Arc::new(ResourceInner {
                attrs,
                schema_url: normalized_schema_url ,
            }),
        }

^ Isn't this easier, as opposed to created a Resource with just attributes, and then cloning it to form a new one with schema_url added ? 

(I might be missing something obvious....)

Reply:
Good comment. Updated accordingly.